integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/template/endpointTemplate/EndpointTemplateSuspensionTest.java

```diff
@@ -0,0 +1,125 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import javax.activation.DataHandler;
+
+/**
+ * Test class to test TemplateEndpoint suspension after retry attempts
+ * Verify fix: https://github.com/wso2/product-ei/issues/1604
+ */
+public class EndpointTemplateSuspensionTest extends ESBIntegrationTest {
+
+    private CarbonAppUploaderClient carbonAppUploaderClient;
+    private ApplicationAdminClient applicationAdminClient;
+    private static String cAppName = "templateEndpointInRegistryTestCapp_1.0.0";
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        //Deploy CAPP
+        carbonAppUploaderClient = new CarbonAppUploaderClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        applicationAdminClient = new ApplicationAdminClient(contextUrls.getBackEndUrl(),getSessionCookie());
+
+        carbonAppUploaderClient.uploadCarbonAppArtifact("templateEndpointInRegistryTestCapp_1.0.0.car",
+                new DataHandler(new URL("file:" + File.separator + File.separator + getESBResourceLocation() +
+                        File.separator + "car" + File.separator + "templateEndpointInRegistryTestCapp_1.0.0.car")));
+
+        boolean cAppDeployed = Utils.isCarFileDeployed(cAppName, applicationAdminClient, 120000);
+        Assert.assertTrue(cAppDeployed, "CApp templateEndpointInRegistryTestCapp_1.0.0.car deployment failed");
+    }
+
+
+    @Test(groups = "wso2.esb", description = "Test invoking EP which cause timeout and verify whether Endpoint "
+            + "template has been marked for suspension")
+    public void testTemplateEndpointSuspension () throws IOException, LogViewerLogViewerException,
+            InterruptedException {
+
+        String contentType = "text/xml";//Content-Type
+        Map<String, String> headers = new HashMap<String, String>();//For HTTP Headers
+        headers.put("Content-Type", contentType);
+        headers.put("SOAPAction", "urn:mediate");
+        String payload =    "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+                            "   <soapenv:Header/>\n" +
+                            "   <soapenv:Body>\n" +
+                            "       <test>test payload</test>\n" +
+                            "   </soapenv:Body>\n" +
+                            "</soapenv:Envelope>";
+
+        LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(),getSessionCookie());
+        logViewerClient.clearLogs();
+
+        log.info("Invoke testEndpointTemplateProxy attempt 1");
+        SimpleHttpClient httpClient1 = new SimpleHttpClient();
+        HttpResponse response1 = httpClient1.doPost(getProxyServiceURLHttp("testEndpointTemplateProxy"), headers, payload, contentType);
+
+        //Wait for timeout markForSuspension retry delay
+        try {
+            Thread.sleep(12000);
```

Comment:
Shall we reduce this time to around 5 sec, as it will reduce the time to run the test suite? 
Also shall we make this a constant for the testcase?

Reply:
+1, will do

Reply:
Fixed,